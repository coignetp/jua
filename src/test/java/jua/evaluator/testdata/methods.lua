local outer = {}
outer.name = 'Mr outer'

local inner = {}
inner.name = 'Mr inner'

outer.inner = inner

function outer:add_outer(v)
    print("called " .. self.name)
    print(self.name)
    self.value = v
end

function outer.inner:add_inner(v)
    print("called " .. self.name)
    self.value = v
end

print("---")
outer:add_outer(5)
print(outer.value)
print(outer.inner.value)
print("---")
outer.inner:add_inner(3)
print(outer.value)
print(outer.inner.value)
print("---")
outer.add_outer(outer.inner, 100)
print(outer.value)
print(outer.inner.value)
print("---")
outer.inner.add_inner(outer, 200)
print(outer.value)
print(outer.inner.value)
print("---")
