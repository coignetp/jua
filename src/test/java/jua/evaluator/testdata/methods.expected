---
called Mr outer
Mr outer
5
nil
---
called Mr inner
5
3
---
called Mr inner
Mr inner
5
100
---
called Mr outer
200
100
---
